crates/runtime/src/init/worker.rs

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use std::sync::Arc;
+
+use crate::{Runtime, metrics, status, timing::TimeMeasurement};
+use ::workers::RouterModel;
+use opentelemetry::KeyValue;
+use snafu::prelude::*;
+
+#[derive(Debug, Snafu)]
+pub enum Error {}
+
+impl Runtime {
+    pub(crate) async fn load_workers(&self) {
+        let app_lock = self.app.read().await;
+
+        if let Some(app) = app_lock.as_ref() {
+            for worker in &app.workers {
+                self.status
+                    .update_worker(&worker.name, status::ComponentStatus::Initializing);
+                self.load_worker(worker).await;
+            }
+        }
+    }
+
+    async fn load_worker(&self, cfg: &spicepod::component::worker::Worker) {
+        let _guard = TimeMeasurement::new(
+            &metrics::workers::LOAD_DURATION_MS,
+            &[KeyValue::new("worker", cfg.name.clone())],
+        );
+
+        tracing::info!("Loading worker [{}]...", cfg.name);
+
+        // Currently, only worker paradigm is [`RouterModel`].
+        let worker = RouterModel::new(cfg.name.clone(), cfg.models.clone(), Arc::clone(&self.llms));
+
+        let mut llm_registry = self.llms.write().await;
+        llm_registry.insert(cfg.name.clone(), Arc::new(worker));
+
+        tracing::info!("Worker [{}] loaded, ready for use", cfg.name);
+        metrics::workers::COUNT.add(1, &[KeyValue::new("worker", cfg.name.clone())]);
+        self.status
+            .update_worker(&cfg.name, status::ComponentStatus::Ready);
+    }
```

Comment:
If we don't need the lock anymore, we should drop it explicitly rather than wait for it to go out of scope since we later call into a function, `update_worker`.